src/transformers/trainer.py

```diff
@@ -272,6 +272,25 @@ def _get_fsdp_ckpt_kwargs():
         return {}
 
 
+def safe_globals():
+    # Starting from version 2.4 PyTorch introduces a stricter check for the objects which
+    # can be loaded with torch.load(). Starting from version 2.6 loading with weights_only=True
+    # requires allowlisting of such objects.
+    # See: https://github.com/pytorch/pytorch/pull/137602
+    # See: https://pytorch.org/docs/stable/notes/serialization.html#torch.serialization.add_safe_globals
+    # See: https://github.com/huggingface/accelerate/pull/3036
+    if version.parse(torch.__version__) < version.parse("2.4.0"):
```

Comment:
just a nit: should it be `"2.6.0"` here or it's really necessary being `"2.4.0"`?

Reply:
Switched to `version < 2.6.0.a0`. Indeed, on switching to context manager I overlooked that it was introduced later. Overall:
* `torch.serialization.add_safe_globals` appeared in pytorch 2.4
* `torch.serialization.safe_globals` (context manager) appeared in 2.5
* And pytorch 2.6 flipped default of `weights_only` in `torch.load` from `False` to `True`

Overall, it indeed does not make sense to have this code working for versions earlier than 2.6 unless we will start calling `torch.load` with explicit `weights_only=True`.

Reply:
Hi! A tiny question: how to get `2.6.0a0` installed. I know how to install night but it gets `dev202411xx` instead of `a0`

Reply:
Anyway, good to use `a0` here for now. Once 2.6 is released, we can change it to `2.6`.

Reply:
I checked. `<2.6.0a0` won't work with nightly. So, I switched to a check I ones spotted in a code by Narsil. This should handle both cases, building from sources and using 2.6 nightly (I checked - works for both on my side):
```
if version.parse(torch.__version__).release < version.parse("2.6").release:
```